### PR TITLE
[p2p] Throttle receive with constant cost function

### PR DESCRIPTION
**Pull request by fxfactorial**
1. remove msgq
2. remove dead top level package values
3. remove unnecessary abstraction/interface/wrapper types
4. Only allow 100 message handlers at a time with a weighted semaphore.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -374,27 +355,60 @@ func (node *Node) AddPendingReceipts(receipts *types.CXReceiptsProof) {
...
 
-func (node *Node) startRxPipeline(
-	receiver p2p.GroupReceiver, queue *msgq.Queue, numWorkers int,
-) {
-	// consumers
-	for i := 0; i < numWorkers; i++ {
-		go queue.HandleMessages(node)
+// Start kicks off the node message handling
+func (node *Node) Start() error {
+	allTopics := node.host.AllTopics()
```
Comment (rlan35) [inline node/node.go:360]: Where are the topics registered?

Comment (fxfactorial) [reply node/node.go:360]: They are already registered done by this point, same as before

Comment (rlan35) [reply node/node.go:360]: Ok, So we don't have topic specific worker pool anymore? Everything is competing for the same resources with equal weight?

Comment (fxfactorial) [reply node/node.go:360]: No, each one is having its own channel - same as before conceptually

Comment (fxfactorial) [reply node/node.go:360]: see the for loop

Comment (rlan35) [reply node/node.go:360]: ok, cool

**Comment by LeoHChen** [comment]
so we expect this PR will prevent OOM issue? with only 300 messages being processed at the same time?

**Comment by fxfactorial** [comment]
> so we expect this PR will prevent OOM issue? with only 300 messages being processed at the same time?

yes. the OOM issue also caused by the msgq impl which held onto references and wouldn't let go of memory. wrt amount of messages - its a throttle which I will make smarter later .

**Pull request merged**